test/unit/ngsiv2/httpBindings-test.js

```diff
@@ -0,0 +1,738 @@
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        },
+        groupCreation = {
+            url: 'http://localhost:4041/iot/services',
+            method: 'POST',
+            json: utils.readExampleFile('./test/groupProvisioning/provisionAliasGroup.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        beforeEach(function(done) {
+
+            contextBrokerMock
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v2/entities/Second%20UL%20Device/attrs' ,
+                utils.readExampleFile('./test/unit/ngsiv2/contextRequests/unprovisionedAliasMeasure2.json'))
+            .reply(204);
+
+            request(groupCreation, function(error, response, body) {
+                request(deviceCreation, function(error, response, body) {
+                    done();
+                    });
+                });
+            });
+
+        it('should use the device preference', function(done) {
+            request(getOptions, function(error, response, body) {
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When a real production request arrives to the IoTA', function() {
+        var postOptions = {
+            url: 'http://localhost:' + config.http.port + '/iot/d',
+            method: 'POST',
+            qs: {
+                i: 'urn:x-iot:smartsantander:u7jcfa:fixed:t311',
+                k: '1234',
+                t: '2016-05-11T10:12:26.476659Z'
+            },
+            headers: {
+                'Content-type': 'text/plain'
+            },
+            body: 'tmp|24.4#hum|58.0#aco|0.1#apa|0.38#ao3|121.0#' +
+            'no2|115.0#pla|43.4551#plo|-3.83381#poa|28.0#spi|0.0#dia|0.0#mit|1492.0#pos|43.4630608,-3.8345434'
+        },
+        provisionProduction = {
+            url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/deviceProvisioning/provisionDeviceProduction.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+        // Note: /v1/updateContext response is not processed by IOTA so its content is irrelevant,
+        // as far as it is a 200 OK
+        beforeEach(function(done) {
+            contextBrokerMock
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v1/updateContext')
+            .reply(200, '{}')
+            .post('/v2/entities/urn:x-iot:smartsantander:u7jcfa:fixed:t311/attrs')
```

Comment:
.post() for this ULR double declaration (L622). Not an expert in nock module, but not sure if this is correct...

Reply:
Yes, this is correct. It means that contextBrokerMock expects first a POST request using the path '/v1/updateContext' (L615) and then another POST request using the path ''/v2/entities/urn:x-iot:smartsantander:u7jcfa:fixed:t311/attrs''. The former corresponds to the call to register the entity in the CB, while the latest corresponds to the call to send observations. This is one option supported by nock call [chaining](https://github.com/node-nock/nock#chaining).

Reply:
Didn't know about nock chaining. Seems quite powerful feature :) Thanks!

NTC